source/transfer_list.rst

```diff
@@ -391,23 +408,60 @@ Inputs:
    `te_base_addr` with `tl.checksum`.
 
 #. If an existing XFERLIST_VOID TE was chosen to be overwritten in step 1, and
-   `old_void_data_size - new_data_size` is greater or equal to `0x8`:
+   `old_void_data_size - new_data_size` is greater or equal to `0x8` then call
+   `Adding a void TE`_ with following arguments:
+
+   #. `void.te.base_addr` = `te_base_addr + align8(new_data_size + 0x8)`
+
+   #. `void.te.size` =  `old_void_data_size - align8(new_data_size)`
+
+Removing a TE
+^^^^^^^^^^^^^
+
+Inputs:
+
+- `tl_base_addr`: Base address of the TL from which TE to be deleted
+- `tag_id`: ID number of the tag to be deleted
+- `scrub_data`: Boolean option to scrub TE data
+
+#. Validate the TL header as mentioned in `Validating a TL header`_
+
+#. Locate the `te_base_addr` of the TE to be removed by iterating through the TL.
+
```

Comment:
To keep the checksum instructions in line with other operations, add here (before making changes):
```
If `has_checksum`, xor the `te.hdr_size + te.data_size` bytes starting at `te_base_addr` with `tl.checksum`
```